src/AwsLambda/AwsLambdaOpenTracer/DataCollector.cs

```diff
@@ -63,23 +65,44 @@ public void TransformSpans(LambdaRootSpan rootSpan)
                 TransactionEvent txnEvent = new TransactionEvent(rootSpan);
                 var errorEvents = _errors.GetAndClearEvents();
                 var errorTraces = _errors.GetAndClearTraces();
-                WriteData(arn, _executionEnv, spans.ToList(), txnEvent, errorEvents, errorTraces);
+
+                var (payload, data) = PreparePayload(arn, _executionEnv, spans.ToList(), txnEvent, errorEvents, errorTraces);
+
+                // If named pipe exists we want to send the payload there instead of writing to standard out.
+                var namedPipe = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, _namedPipePath));
+                if (File.Exists(namedPipe))
+                {
+                    WriteToNamedPipe(payload, namedPipe);
+                    return;
+                }
+                WriteData(payload, data);
```

Comment:
I don't know how norms are in C#, but I generally find else blocks clearer than early returns in cases like this.